compiler/src/yul/runtime/abi.rs

```diff
@@ -0,0 +1,105 @@
+use crate::errors::CompileError;
```

Comment:
there may not be a nice way to do this but i'll just point out it would be ideal to avoid duplicating the logic of making the selector in the ABI and also here...

Reply:
agreed. might justify a utils module